bigtable/google/cloud/bigtable/app_profile.py

```diff
@@ -0,0 +1,309 @@
+        app_profile_id = match_app_profile_name.group('app_profile_id')
+
+        result = cls(app_profile_id, instance)
+        result._update_from_pb(app_profile_pb)
+        return result
+
+    def _update_from_pb(self, app_profile_pb):
+        """Refresh self from the server-provided protobuf.
+        Helper for :meth:`from_pb` and :meth:`reload`.
+        """
+        self.routing_policy_type = None
+        self.allow_transactional_writes = None
+        self.cluster_id = None
+
+        self.description = app_profile_pb.description
+
+        routing_policy_type = None
+        if app_profile_pb.HasField('multi_cluster_routing_use_any'):
+            routing_policy_type = RoutingPolicyType.ANY
+            self.allow_transactional_writes = False
+        else:
+            routing_policy_type = RoutingPolicyType.SINGLE
+            self.cluster_id = app_profile_pb.single_cluster_routing.cluster_id
+            self.allow_transactional_writes = (
+                app_profile_pb.single_cluster_routing
+                .allow_transactional_writes)
+        self.routing_policy_type = routing_policy_type
+
+    def reload(self):
+        """Reload the metadata for this cluster"""
+
+        app_profile_pb = (
+            self._instance._client.instance_admin_client.get_app_profile(
+                self.name))
+
+        # NOTE: _update_from_pb does not check that the project and
+        #       app_profile ID on the response match the request.
+        self._update_from_pb(app_profile_pb)
+
+    def exists(self):
+        """Check whether the app profile already exists.
+
+        :rtype: bool
+        :returns: True if the app profile exists, else False.
+        """
+        client = self._instance._client
+        try:
+            client.instance_admin_client.get_app_profile(self.name)
+            return True
+        # NOTE: There could be other exceptions that are returned to the user.
+        except NotFound:
+            return False
+
+    def create(self, ignore_warnings=None):
+        """Create this app profile.
+
+        .. note::
+
+            Uses the ``project``, ``instance`` and ``app_profile_id`` on the
+            current :class:`AppProfile` in addition to the ``serve_nodes``.
+            To change them before creating, reset the values via
+
+            .. code:: python
+
+                cluster.serve_nodes = 8
+                cluster.cluster_id = 'i-changed-my-mind'
+
+            before calling :meth:`create`.
+
+        :type: ignore_warnings: bool
+        :param: ignore_warnings: (Optional) If true, ignore safety checks when
+                                 creating the app profile.
+        """
+        if not self.routing_policy_type:
+            raise ValueError('AppProfile required routing policy.')
+
+        single_cluster_routing = None
+        multi_cluster_routing_use_any = None
+
+        if self.routing_policy_type == RoutingPolicyType.ANY:
```

Comment:
Both `update` and `create` need this logic.  Please create a single method that returns a tuple.  Or even better, just create an `AppProfile`.